pkg/upgrade/upgrade.go

```diff
@@ -652,6 +652,10 @@ func cleanupNimIntegration(ctx context.Context, cli client.Client, oldRelease cl
 
 		for _, delObj := range deleteObjs {
 			if gErr := cli.Get(ctx, types.NamespacedName{Name: delObj.name, Namespace: applicationNS}, delObj.obj); gErr != nil {
+				if errors.Is(gErr, &meta.NoKindMatchError{}) {
+					log.V(1).Error(k8serr.NewNotFound(schema.GroupResource{}, delObj.obj.GetName()), fmt.Sprintf("%s %s not found", delObj.desc, delObj.name))
```

Comment:
is this actually an error ? if neither kserve, nor modelmesh have never been installed before, this would result in an error but it is not (IMHO)

Reply:
but this is a check on the CRD, not on object, right?
i think the error was complaining not seeing "Template" in the cluster.

Reply:
Yes it would signal that the kind is unknown. My point is - if this should be seen as an error or not because if the type do not exist, then also the object won't exist. 

However if the `Template` type is not know, would the installation of the component fail ?

Reply:
> However if the `Template` type is not know, would the installation of the component fail ?

if it fails, let it fail there in the component, not on cleanup of some leftover :)

Reply:
reverted this change